crates/miden-objects/src/note/note_tag.rs

```diff
@@ -248,23 +270,55 @@ impl fmt::Display for NoteTag {
 
 impl From<u32> for NoteTag {
     fn from(value: u32) -> Self {
-        Self(value)
+        let prefix = value >> 30;
+        let lower_30_bits = value & 0x3FFFFFFF; // Mask to get the lower 30 bits
+
+        match prefix {
+            0b00 => {
+                // Network, Specific Account (lower 30 bits are the account ID)
+                Self::NetworkAccount(lower_30_bits)
+            },
```

Comment:
I think this should become `TryFrom<u32> for NoteTag` because we should return `NoteError::NetworkExecutionRequiresNetworkAccount` if `NoteTag::NetworkAccount` contains a bit pattern whose least significant bit is `0`, indicating that that account ID is _not_ a network account.